CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.6)
+cmake_minimum_required(VERSION 2.8.12)
```

Comment:
Do we need any CMake features that require this update? If not, I don't think we should update this.